invoiceregistry.go

```diff
@@ -83,6 +89,10 @@ func (i *invoiceState) totalSetAmt() int {
 	return total
 }
 
+func (i *invoiceState) isSetComplete() bool {
+	return i.totalSetAmt() == int(i.invoice.Value)
```

Comment:
The current implementation rejects overpayments intentionally, to avoid having to deal with the unexpected extra sats. An application connected to multiplexer needs to be able to deal with overpayment too, it needs to be accounted for properly, etc.

It could be a config setting though, whether to accept overpayments or not. Or when auto-settle is disabled, the application could inspect the actually paid value during processing of the `ACCEPTED` event and reject if it doesn't want to handle overpayments.

Then there is also the case of zero valued invoices, which are not yet supported either.

A different, even more minimalistic approach could be to not encode expiry and amount in the payment address at all. Just pass every payment on to the application for deciding whether to settle the payment or not. Perhaps that's a bit too extreme though. And also it won't work with auto-settle.